ff/serving/metadata-rest/metadata-rest-server.go

```diff
@@ -0,0 +1,611 @@
+		}
+	}
+	return variantMap, nil
+}
+
+func (m MetadataServer) readFromTrainingSet(trainingSet *metadata.TrainingSet) (map[string]TrainingSetVariantResource, *FetchError) {
+	variantMap := make(map[string]TrainingSetVariantResource)
+	variants, err := trainingSet.FetchVariants(m.client, context.Background())
+	if err != nil {
+		fetchError := &FetchError{StatusCode: 500, Type: "training set variants"}
+		m.logger.Errorw("Failed to fetch variants", "Error", err)
+		return variantMap, fetchError
+	}
+	for _, variant := range variants {
+		variantMap[variant.Variant()] = TrainingSetVariantResource{
+			Created:     variant.Created(),
+			Description: variant.Description(),
+			Name:        variant.Name(),
+			Variant:     variant.Name(),
+			Owner:       variant.Owner(),
+			Provider:    variant.Provider(),
+			Label:       variant.Label(),
+			Features:    variant.Features(),
+		}
+	}
+	return variantMap, nil
+}
+
+func (m MetadataServer) readFromSource(source *metadata.Source) (map[string]SourceVariantResource, *FetchError) {
+	variantMap := make(map[string]SourceVariantResource)
+	variants, err := source.FetchVariants(m.client, context.Background())
+	if err != nil {
+		fetchError := &FetchError{StatusCode: 500, Type: "source variants"}
+		m.logger.Errorw("Failed to fetch variants", "Error", err)
+		return variantMap, fetchError
+	}
+	for _, variant := range variants {
+		variantMap[variant.Variant()] = SourceVariantResource{
+			Created:      variant.Created(),
+			Description:  variant.Description(),
+			Name:         variant.Name(),
+			Type:         variant.Type(),
+			Variant:      variant.Name(),
+			Owner:        variant.Owner(),
+			Provider:     variant.Provider(),
+			Labels:       variant.Labels(),
+			Features:     variant.Features(),
+			TrainingSets: variant.TrainingSets(),
+		}
+	}
+	return variantMap, nil
+}
+
+func (m MetadataServer) readFromLabel(label *metadata.Label) (map[string]LabelVariantResource, *FetchError) {
+	variantMap := make(map[string]LabelVariantResource)
+	variants, err := label.FetchVariants(m.client, context.Background())
+	if err != nil {
+		fetchError := &FetchError{StatusCode: 500, Type: "label variants"}
+		m.logger.Errorw("Failed to fetch variants", "Error", err)
+		return variantMap, fetchError
+	}
+	for _, variant := range variants {
+		variantMap[variant.Variant()] = LabelVariantResource{
+			Created:      variant.Created(),
+			Description:  variant.Description(),
+			Entity:       variant.Entity(),
+			Name:         variant.Name(),
+			Type:         variant.Type(),
+			Variant:      variant.Name(),
+			Owner:        variant.Owner(),
+			Provider:     variant.Provider(),
+			Source:       variant.Source(),
+			TrainingSets: variant.TrainingSets(),
+		}
+	}
+	return variantMap, nil
+}
+
+func (m MetadataServer) GetMetadata(c *gin.Context) {
+	switch c.Param("type") {
+	case "features":
+		features, err := m.client.GetFeatures(context.Background(), []string{c.Param("resource")})
```

Comment:
If client doesn't have a GetFeature( function, I'd add one and use that, makes the code a little cleaner

Reply:
Currently no singular getfeature function or likewise for other resources.

Reply:
Could you add one, it's essentially the same logic as here